src/cert_context.rs

```diff
@@ -119,6 +124,43 @@ impl CertContext {
         }
     }
 
+    /// Get certificate as PEM-formatted X509 certificate.
+    pub fn to_pem(&self) -> io::Result<String> {
+        unsafe {
+            let mut len = 0;
+            let ok = crypt32::CryptBinaryToStringA(
+                (*self.0).pbCertEncoded,
+                (*self.0).cbCertEncoded,
+                0 as winapi::DWORD,
+                ptr::null_mut(),
+                &mut len,
+            );
+            if ok != winapi::TRUE {
+                return Err(io::Error::last_os_error());
+            }
+
+            let mut buf = vec![0u8; len as usize];
+            let ok = crypt32::CryptBinaryToStringA(
+                (*self.0).pbCertEncoded,
+                (*self.0).cbCertEncoded,
+                0 as winapi::DWORD,
+                buf.as_mut_ptr() as *mut i8,
+                &mut len,
+            );
+            if ok != winapi::TRUE {
+                return Err(io::Error::last_os_error());
+            }
+
+            buf.pop();
```

Comment:
Is this a newline or something?

Reply:
Looks like you can pass `CRYPT_STRING_NOCRLF` to disable that.

Reply:
It's a trailing \0 - I'll see if it looks cleaner with CStr() if you like?

Reply:
Ah, gotcha. I think it's fine with the pop if you stick a comment in there saying that's what's going on.